### PR TITLE
HIPP-233: LDAP Stub

### PR DESCRIPTION
**Pull request by PaulCDurham**
https://jira.tools.tax.service.gov.uk/browse/HIPP-233

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,21 @@ Once everything is up and running you can access the application at
...
+* internal-auth-frontend has a stub that will run locally only
+* our own stub hosted on api-hub-frontend which can run locally and in all deployed non-production environments
+
+Locally the service is configured to use:
+* our own stub when running in dev mode
+* the internal-auth-frontend stub otherwise
+
+The configuration for local internal-auth-frontend is in `application.conf`.
+
+The configuration for our own stub in dev mode is in the `PlayKeys.devSettings`
```
Comment (laurahk) [inline README.md:40]: could you add the related commands of `sbt start` above and `sbt run` here

File: app/models/user/UserModel.scala [hunk 2]
```diff
@@ -16,4 +16,10 @@
 
 package models.user
 
-case class UserModel(userId: String, email: Option[String] = Option.empty)
+import viewmodels.WithName
+
+sealed trait UserType
```
Comment (laurahk) [inline app/models/user/UserModel.scala:21]: I'd move UserType to it's own class so it's easier to find/open in IntelliJ. People have differing opinions on related case classes though so just my personal preference.

File: app/testonly/Retrievals.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+package testonly
+
+import play.api.libs.json.{Format, Json}
+
+case class Retrievals(retrievals: Seq[String])
+
+object Retrievals {
+
+  def apply(principal: String, email: Option[String]): Retrievals = {
+    Retrievals(Seq(principal, email.getOrElse(s"$principal@digital.hmrc.gov.uk")))
```
Comment (laurahk) [inline app/testonly/Retrievals.scala:26]: This functionality of creating the email from the username has been discussed but I can't find the AC for it on this set of tickets (I could have overlooked it though). Please can you find the related ticket / talk to Whitney / update the ACs.

Comment (laurahk) [reply app/testonly/Retrievals.scala:26]: Also is there a test for this?

Comment (PaulCDurham) [reply app/testonly/Retrievals.scala:26]: This is in the test only bit of code that pretends to be auth. I doubt that is or should be mentioned in the ACs.

Comment (laurahk) [reply app/testonly/Retrievals.scala:26]: Apologises - I just realised this is stub code!

**Review by laurahk — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (PaulCDurham) [reply README.md:40]: Done

**Review by laurahk — APPROVED** [review]

**Pull request merged**